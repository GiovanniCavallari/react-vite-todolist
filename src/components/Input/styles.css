.input-container {
  width: 100%;
  display: flex;
  align-items: flex-start;
  justify-content: center;
  flex-direction: column;
}

.input-label {
  width: 100%;
  margin: 0 0 4px;
}

.input {
  width: 100%;
  border: 1px solid #aaa;
  padding: 12px 8px;
  border-radius: 4px;
}
