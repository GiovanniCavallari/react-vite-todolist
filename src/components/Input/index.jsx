import React from 'react';
import P from 'prop-types';

import './styles.css';

function Input({ name, label, value, placeholder, onChange }) {
  return (
    <div className="input-container">
      <label className="input-label">{label}</label>
      <input
        className="input"
        type="text"
        name={name}
        value={value}
        onChange={onChange}
        placeholder={placeholder}
      />
    </div>
  );
}

Input.propTypes = {
  name: P.string.isRequired,
  label: P.string.isRequired,
  value: P.string.isRequired,
  onChange: P.func.isRequired,
  placeholder: P.string,
};

export default Input;
