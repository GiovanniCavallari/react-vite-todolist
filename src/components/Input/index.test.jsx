import { render, fireEvent } from '@testing-library/react';
import Input from './index';

describe('Test Input Component', () => {
  const onChange = jest.fn();

  const component = (
    <Input
      label="label"
      name="name"
      placeholder="placeholder"
      value=""
      onChange={onChange}
    />
  );

  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const { asFragment, getByText, getByPlaceholderText } = render(component);
    const input = getByPlaceholderText('placeholder');
    const label = getByText('label');

    expect(asFragment).toMatchSnapshot();
    expect(input).toBeInTheDocument();
    expect(label).toBeInTheDocument();
  });

  it('should change value and call onChange prop', () => {
    const { getByPlaceholderText } = render(component);
    const element = getByPlaceholderText('placeholder');

    fireEvent.change(element, {
      target: { value: 'value' },
    });

    expect(onChange).toBeCalled();
  });
});
