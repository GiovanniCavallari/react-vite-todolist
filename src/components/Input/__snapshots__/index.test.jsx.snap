// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Input Component should render correctly 1`] = `[Function]`;
